tests/unit/test_charm_non_leader_unit.py

```diff
@@ -0,0 +1,224 @@
+                "alertmanager-configurer": {
+                    "override": "replace",
+                    "startup": "enabled",
+                    "command": f"alertmanager_configurer "
+                    f"-port={test_alertmanager_configurer_port} "
+                    f"-alertmanager-conf={TEST_ALERTMANAGER_CONFIG_FILE} "
+                    "-alertmanagerURL="
+                    f"{test_dummy_http_server_host}:{test_dummy_http_server_port} "
+                    f"-multitenant-label={TEST_MULTITENANT_LABEL} "
+                    "-delete-route-with-receiver=true ",
+                }
+            }
+        }
+
+        self.harness.container_pebble_ready("alertmanager-configurer")
+
+        updated_plan = self.harness.get_container_pebble_plan("alertmanager-configurer").to_dict()
+        self.assertEqual(expected_plan, updated_plan)
+
+    def test_given_dummy_http_server_container_ready_when_pebble_ready_then_pebble_plan_is_updated_with_correct_pebble_layer(  # noqa: E501
+        self,
+    ):
+        expected_plan = {
+            "services": {
+                "dummy-http-server": {
+                    "override": "replace",
+                    "startup": "enabled",
+                    "command": "nginx",
+                }
+            }
+        }
+        self.harness.container_pebble_ready("dummy-http-server")
+
+        updated_plan = self.harness.get_container_pebble_plan("dummy-http-server").to_dict()
+        self.assertEqual(expected_plan, updated_plan)
+
+    @patch("charm.AlertmanagerConfigDirWatcher", Mock())
+    def test_given_alertmanager_relation_created_and_alertmanager_configurer_container_ready_when_pebble_ready_then_charm_goes_to_active_state(  # noqa: E501
+        self,
+    ):
+        self.harness.add_relation("alertmanager", "alertmanager-k8s")
+        self.harness.container_pebble_ready("dummy-http-server")
+
+        self.harness.container_pebble_ready("alertmanager-configurer")
+
+        assert self.harness.charm.unit.status == ActiveStatus()
+
+    @patch("ops.model.Container.push")
+    @patch(f"{ALERTMANAGER_CLASS}.ALERTMANAGER_DEFAULT_CONFIG", new_callable=PropertyMock)
+    @patch(f"{ALERTMANAGER_CLASS}.ALERTMANAGER_CONFIG_FILE", new_callable=PropertyMock)
+    @patch("charm.AlertmanagerConfigDirWatcher", Mock())
+    def test_given_alertmanager_default_config_when_start_then_alertmanager_config_is_created_using_default_data(  # noqa: E501
+        self, patched_alertmanager_config_file, patched_alertmanager_default_config, patched_push
+    ):
+        patched_alertmanager_config_file.return_value = TEST_ALERTMANAGER_CONFIG_FILE
+        patched_alertmanager_default_config.return_value = TEST_ALERTMANAGER_DEFAULT_CONFIG
+        expected_config_push_calls = [
+            call(TEST_ALERTMANAGER_CONFIG_FILE, TEST_ALERTMANAGER_DEFAULT_CONFIG)
+        ]
+
+        self.harness.charm.on.start.emit()
+
+        patched_push.assert_has_calls(expected_config_push_calls)
```

Comment:
If we only expect 1 call, it's preferable to use `assert_called_with()`. Assert has calls is mostly useful when you expect a method to be called more than once.

Reply:
In this case I'm using it `has_calls`, because push is called by other functions too, so there's more calls in the bucket. I could probably do `expected_config_push_calls in pached_push.mock_calls`

Reply:
`assert_any_call` in this case can be used. `patched_push.assert_any_call(TEST_ALERTMANAGER_CONFIG_FILE, TEST_ALERTMANAGER_DEFAULT_CONFIG)`

Reply:
True, updated.